src/main/java/net/dv8tion/jda/internal/managers/GuildScheduledEventManagerImpl.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.managers;
+
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.managers.GuildScheduledEventManager;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.requests.Route;
+import net.dv8tion.jda.internal.utils.Checks;
+import okhttp3.RequestBody;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+public class GuildScheduledEventManagerImpl extends ManagerBase<GuildScheduledEventManager> implements GuildScheduledEventManager
+{
+    protected GuildScheduledEvent event;
+    protected String name, description;
+    protected long channelId;
+    protected String location;
+    protected Icon image;
+    protected OffsetDateTime startTime, endTime;
+    protected int entityType;
+    protected GuildScheduledEvent.Status status;
+
+    public GuildScheduledEventManagerImpl(GuildScheduledEvent event)
+    {
+        super(event.getJDA(), Route.Guilds.MODIFY_SCHEDULED_EVENT.compile(event.getGuild().getId(), event.getId()));
+        this.event = event;
+        if (isPermissionChecksEnabled())
+            checkPermissions();
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEvent getGuildScheduledEvent()
+    {
+        GuildScheduledEvent realEvent = event.getGuild().getScheduledEventById(event.getIdLong());
+        if (realEvent != null)
+            event = realEvent;
+        return event;
+    }
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventManagerImpl setName(@Nonnull String name)
+    {
+        Checks.notBlank(name, "Name");
```

Comment:
Returning to the blank name from before, you check it, but you didn't document it!

Reply:
* @throws java.lang.IllegalArgumentException
     *         If the new name is empty, {@code null}, or contains more than {@value GuildScheduledEvent#MAX_NAME_LENGTH}
     *         characters
     
     to me empty = blank or null so its included in the docs, no ?

Reply:
this comment can be marked as resolved, its quite hard to group stuff together, that goes together.

Reply:
Blank may be a better choice in this scenario considering [String#isEmpty](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#isEmpty--) returns true only if the length of the string is zero, while in Java 11 and forward [String#isBlank](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#isBlank()) also makes sure that the string doesn't contain only whitespaces, which is also how Checks#notBlank operates.